Fix Tidy UI compatibility with Foundry VTT v13

digennarot opened the pull request:
Fix Tidy UI compatibility with Foundry VTT v13
This update resolves several compatibility issues that prevented the Tidy UI module from working properly in Foundry VTT v13:

- Fixed HTML parameter handling - Added proper conversion logic for DOM elements vs jQuery objects in hook callbacks
- Improved jQuery selector scope - Fixed selectors that weren't properly scoped to avoid UI conflicts
- Updated clipboard functionality - Replaced deprecated document.execCommand() with modern navigator.clipboard.writeText() API
- Enhanced JSON validation - Replaced regex-based validation with proper JSON.parse() for better reliability
- Fixed HTML structure - Corrected missing closing tags and improved markup
- Better error handling - Added proper error handling for file import operations
- Memory leak prevention - Added array resets to ensure clean state between operations

digennarot committed:
Update tidy-ui_game-settings.js

Lexicality commented on src/tidy-ui_game-settings.js:
```diff
@@ -3,8 +3,12 @@ var expandedModules = [];
 Hooks.on("renderSettingsConfig", (app, html) => {
-  // wrap separat module settings
-  html
+  // Convert jQuery object to DOM element if needed
+  const htmlElement = html instanceof jQuery ? html[0] : html;
+  const $html = htmlElement instanceof Element ? $(htmlElement) : $(html);
```
You don't need to do this, you can just do `html = $(html)` since calling `$()` on a `jQuery` object is a no-op.

digennarot replied:
thanks for letting me know I patched it, can you review it now?

Lexicality replied:
I think it'd be a lot easier to read the diff if you hadn't renamed all the variables, but that does fix my comment

digennarot committed:
Update tidy-ui_game-settings.js

Replaced manual jQuery object normalization with a single `$(html)` call.
The jQuery constructor is a no-op for existing jQuery objects, making
the previous verbose checks redundant.

digennarot committed:
Update tidy-ui_game-settings.js

skellitor301 commented:
Is there a way to get Foundry to pull this fork of the module? I'm having trouble. It says could not find module manifest in archive.

digennarot commented:
> Is there a way to get Foundry to pull this fork of the module? I'm having trouble. It says could not find module manifest in archive.

go directly to the raw https://raw.githubusercontent.com/digennarot/tidy-ui_game-settings/refs/heads/master/module.json
